internal/cli/cmd/version.go

```diff
@@ -32,6 +37,38 @@ var (
 	Version string
 )
 
+type CiliumVersion struct {
+	client *k8s.Client
+}
+
+func NewCiliumVersion() *CiliumVersion {
+	return &CiliumVersion{client: k8sClient}
+}
+
+func (c *CiliumVersion) GetRunningCiliumVersionForCluster(ctx context.Context) (string, error) {
+	nss, err := c.client.ListNamespaces(ctx, metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("unable to list k8s namespace: %w", err)
+	}
+
+	// First look for ns kube-system
+	version, _ := c.client.GetRunningCiliumVersion(ctx, "kube-system")
+	if version != "" {
+		return version, nil
+	}
+
+	if len(nss.Items) > 0 {
+		for _, namespace := range nss.Items {
+			version, _ := c.client.GetRunningCiliumVersion(ctx, namespace.Name)
+			if version != "" {
+				return version, nil
+			}
+		}
+	}
+
+	return "", errors.New("unable to obtain cilium version: no cilium pods found")
+}
+
```

Comment:
Instead of first trying `kube-system` and potentially looping over multiple namespaces, how about just adding a `--namespace` flag to `cilium version` with a default set to `kube-system`, so it will work in the same way as other commands taking a namespace?

Reply:
Yeah,that sounds like a good suggestion